smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/DocumentMemberSerVisitor.java

```diff
@@ -83,21 +86,70 @@ public DocumentMemberSerVisitor(
             String dataSource,
             Format defaultTimestampFormat
     ) {
+        this(context, dataSource, defaultTimestampFormat, SetUtils.of());
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param context The generation context.
+     * @param dataSource The in-code location of the data to provide an input of
+     *                   ({@code input.foo}, {@code entry}, etc.)
+     * @param defaultTimestampFormat The default timestamp format used in absence
+     *                               of a TimestampFormat trait.
+     * @param additionalParams A set of variable names to be passed as additional
+     *                         parameters to any delegate deserializers.
+     */
+    public DocumentMemberSerVisitor(
+            GenerationContext context,
+            String dataSource,
+            Format defaultTimestampFormat,
+            Set<String> additionalParams
+    ) {
+        this.additionalParams = additionalParams;
         this.context = context;
         this.dataSource = dataSource;
         this.defaultTimestampFormat = defaultTimestampFormat;
     }
 
+    /**
+     * Gets the generation context.
+     *
+     * @return The generation context.
+     */
+    protected final GenerationContext getContext() {
+        return context;
+    }
+
     /**
      * Gets the in-code location of the data to provide an input of
      * ({@code input.foo}, {@code entry}, etc.).
      *
      * @return The data source.
      */
-    protected String getDataSource() {
+    protected final String getDataSource() {
         return dataSource;
     }
 
+    /**
+     * Gets the default timestamp format used in absence of a TimestampFormat trait.
+     *
+     * @return The default timestamp format.
+     */
+    protected final Format getDefaultTimestampFormat() {
+        return defaultTimestampFormat;
+    }
+
+    /**
+     * Gets a set of variable names to be passed as additional parameters
+     * to any delegate serializers.
```

Comment:
The same not as earlier: this needs to document some kind of ordering guarantee (insertion order, alphabetical, etc)